src/pipe.js

```diff
@@ -74,18 +86,56 @@ var Pipe = (function (doc, perf) { //eslint-disable-line no-unused-vars, strict
                     };
                 }
 
+                function isMainContentInteractive() {
+                    for (var index in mainFragments) {
+                        var obj = mainFragments[index];
+                        if (!obj.initialized) {
+                            return false;
+                        }
+                    }
+                    return true;
+                }
+
+                function measureInteractivity() {
+                    // Handle if there are no main fragments on the page
+                    if (JSON.stringify(mainFragments) === '{}') {
+                        isInteractiveDone = true;
+                        return;
+                    }
+
+                    if (isMainContentInteractive()) {
+                        isInteractiveDone = true;
+                        // This will measure the high resolution time from navigationStart till current time
+                        perf.measure('interactive');
+                    }
+                }
+
+                function markInitialized() {
+                    if (typeof mainFragments[index] === 'object') {
+                        mainFragments[index].initialized = true;
+                    }
+                }
+
                 function doInit(init, node, callback) {
                     var fragmentRender = init(node);
+                    var handlerFn = function() {
+                        markInitialized();
+                        callback();
+                        // Measure the main content interactivity once all the main fragments are initialized
+                        if (!isInteractiveDone) {
+                            measureInteractivity();
+                        }
+                    };
                     // Check if the response from fragment is a Promise to allow lazy rendering
                     if (isPromise(fragmentRender)) {
-                        fragmentRender.then(callback, callback);
+                        fragmentRender.then(handlerFn, handlerFn);
```

Comment:
might be nicer to use the `.catch(handlerFn)` form instead, making it explicit that this fn handles all cases.